adam/perception/perception_graph.py

```diff
@@ -313,111 +328,132 @@ def from_schema(object_schema: ObjectStructuralSchema) -> "PerceptionGraphPatter
         """
         Creates a pattern for recognizing an object based on its *object_schema*.
         """
-        graph = DiGraph()
-        object_to_node: Dict[Any, NodePredicate] = {}
-
-        PerceptionGraphPattern._translate_schema(
-            # since the root object does not itself have a `SubObject` object,
-            # we wrap it in one.
-            object_=SubObject(object_schema),
-            object_schema=object_schema,
-            graph=graph,
-            schema_node_to_pattern_node=object_to_node,
+
+        # First, we generate a PerceptionGraph corresponding to this schema
+        # by instantiating a situation which contains a single object
+        # of the desired type.
+        schema_situation_object = SituationObject.instantiate_ontology_node(
+            ontology_node=object_schema.ontology_node, ontology=GAILA_PHASE_1_ONTOLOGY
+        )
+        situation = HighLevelSemanticsSituation(
+            ontology=GAILA_PHASE_1_ONTOLOGY, salient_objects=[schema_situation_object]
+        )
+        perception_generator = HighLevelSemanticsSituationToDevelopmentalPrimitivePerceptionGenerator(
+            GAILA_PHASE_1_ONTOLOGY
+        )
+        perception = perception_generator.generate_perception(
+            situation, chooser=RandomChooser.for_seed(0)
         )
-        return PerceptionGraphPattern(graph)
+        perception_graph = PerceptionGraph.from_frame(
+            first(perception.frames)
+        ).copy_as_digraph()
+
+        # We remove certain information that is added by situation generation, but is not in schema
+        # Removed in order: colors, ontology nodes (properties), regions related to ground, ground, learner and
+        # finally any leftover islands that dont belong to the schema's graph component
+        nodes_to_remove = [
```

Comment:
Actually, can we simplify this whole things to:

* delete everything not reachable (taking into account edge direction) from the root object perception
* From that, delete colors.